selfservice/flow/registration/hook.go

```diff
@@ -82,6 +85,37 @@ func NewHookExecutor(d executorDependencies) *HookExecutor {
 	return &HookExecutor{d: d}
 }
 
+func (e *HookExecutor) handleRegistrationError(_ http.ResponseWriter, r *http.Request, ct identity.CredentialsType, f *Flow, i *identity.Identity, flowError error) error {
+	if f != nil {
+		if i != nil {
+			var group node.Group
+			switch ct {
+			case identity.CredentialsTypePassword:
+				group = node.PasswordGroup
+			case identity.CredentialsTypeOIDC:
+				group = node.OpenIDConnectGroup
```

Comment:
Can you please extract this logic into a common helper function that converts from credential types to groups and vice versa? And then use this instead of having the correct group conversion applied in different files?